app/lib/task/backend.dart

```diff
@@ -741,56 +750,75 @@ class TaskBackend {
   /// The returned [BlobIndex] will carry a [BlobIndex.blobId] that is the
   /// path for the blob being reference, this path will include runtime-version,
   /// package name, version and randomized blobId.
-  Future<BlobIndex?> _taskResultIndex(String package, String version) async =>
-      await cache.taskResultIndex(package, version).get(() async {
-        // Don't try to load index if we don't consider the version for analysis.
-        final status = await packageStatus(package);
-        if (!status.versions.containsKey(version)) {
-          return BlobIndex.empty(blobId: '');
+  Future<BlobIndex?> _taskResultIndex(
+    String package,
+    String version, {
+    bool skipCached = false,
+  }) async {
```

Comment:
```dart
extension NeatCacheEntry<T> on Entry<T> {
  // Get or create a value.
  Future<T> obtain(
    Future<T> Function() create, {
    Duration? ttl,
    bool purgeCache = false,
  }) async {
    if (purgeCache) {
      final value = create();
      if (ttl != null) {
        await set(value, ttl);
      } else {
        await set(value);
      }
      return value;
    }
    if (ttl != null) {
      return (await get(create, ttl))!;
    }
    return (await get(create))!;
  }
}
```

Filed https://github.com/google/dart-neats/issues/214 suggesting we add this upstream.